src/control/mobile_manipulation/mobile_manipulation_learner.py

```diff
@@ -0,0 +1,247 @@
+from stable_baselines3.common.vec_env import VecEnv
+from stable_baselines3.sac import SAC
+from typing import Optional
+from urllib.request import urlretrieve
+
+from control.mobile_manipulation.mobileRL.evaluation import evaluation_rollout
+from control.mobile_manipulation.mobileRL.stablebl_callbacks import MobileRLEvalCallback
+from engine.constants import OPENDR_SERVER_URL
+from engine.learners import LearnerRL
+
+
+class MobileRLLearner(LearnerRL):
+    def __init__(self, env, lr=1e-5, iters=1_000_000, batch_size=64, lr_schedule='linear',
+                 lr_end: float = 1e-6, backbone='MlpPolicy', checkpoint_after_iter=20_000, checkpoint_load_iter=0,
+                 restore_model_path: Optional[str] = None, temp_path='', device='cuda', seed: int = None,
+                 buffer_size: int = 100_000, learning_starts: int = 0, tau: float = 0.001, gamma: float = 0.99,
+                 explore_noise: float = 0.5, explore_noise_type='normal', ent_coef='auto', nr_evaluations: int = 50,
+                 evaluation_frequency: int = 20_000):
+        """
+        Specifies a soft-actor-critic (SAC) agent that can be trained for mobile manipulation.
+        Internally uses Stable-Baselines3 (https://github.com/DLR-RM/stable-baselines3).
+        """
+        super(LearnerRL, self).__init__(lr=lr, iters=iters, batch_size=batch_size, optimizer='adam',
+                                        lr_schedule=lr_schedule, backbone=backbone, network_head='',
+                                        checkpoint_after_iter=checkpoint_after_iter,
+                                        checkpoint_load_iter=checkpoint_load_iter, temp_path=temp_path,
+                                        device=device, threshold=0.0, scale=1.0)
+        self.seed = seed
+        self.lr_end = lr_end
+        self.nr_evaluations = nr_evaluations
+        self.evaluation_frequency = evaluation_frequency
+        self.stable_bl_agent = self._construct_agent(env=env,
+                                                     buffer_size=buffer_size,
+                                                     learning_starts=learning_starts,
+                                                     tau=tau,
+                                                     gamma=gamma,
+                                                     explore_noise=explore_noise,
+                                                     explore_noise_type=explore_noise_type,
+                                                     ent_coef=ent_coef)
+
+        if checkpoint_load_iter:
+            if restore_model_path == 'pretrained':
+                assert checkpoint_load_iter == 1_000_000, "pretrained models are provided for step 1_000_000"
+            else:
+                restore_model_path = os.path.join(restore_model_path, f"model_step{checkpoint_load_iter}")
+            self.load(restore_model_path)
+
+    @staticmethod
+    def _download_pretrained(save_path: str, robot_name: str):
```

Comment:
Even though the signature of this function has not been specified in the first version of specifications, I think it would be nice if we could stick to an existing way of implementing this (e.g., https://github.com/tasostefas/opendr_internal/blob/master/src/opendr/perception/pose_estimation/lightweight_open_pose/lightweight_open_pose_learner.py#L906 ) in order to avoid having to update this later. So, it would be very nice if you could rename and add any arguments (if needed) to follow the function signature provided by the rest of the implementations.

Reply:
This is a private helper function as indicated by the underscore. The load() function adheres to the BaseLearner Api and is the one that should be called by any user. I am not sure how making this public would provide additional functionality over the load function and in the worst case seems to me like it would confuse users, as there is also no public download functionality in any of the base classes?

Reply:
What we have found out during the implementation is that even though there is no public *download()* function specified in D2.1, most method needs one to download a pretrained model. If this is the case and you download a pre-trained model, then it would be beneficial to provide a public *download()* method to follow the interface of the rest of the methods (even through it is not included in the base learner). This is indeed not yet specified in D2.1, but it will most probably be in the next version, since virtually all methods provide this functionality.

Reply:
I see. I have adjusted the signature.